src/components/utils/test/timer_test.cc

```diff
@@ -43,214 +43,276 @@ namespace components {
 namespace timer_test {
 namespace {
 
-sync_primitives::Lock test_lock;
-sync_primitives::ConditionalVariable lock_monitor;
+sync_primitives::Lock shot_lock;
+sync_primitives::ConditionalVariable shot_condition;
+
+const std::string kTimerName = "TestTimer";
+const bool kSingleShot = true;
 const uint32_t kDefaultTimeout = 30u;
-const std::string kTimerName = "test_timer";
+const uint32_t kDefaultTimeoutRestart = 45u;
```

Comment:
@OHerasym these timeout values are both got from ceiling :smile:. We just need two different values. Good question is about different hardware and system performance. I mean: is this timeout enough for some slow machines? Well, I cannot guarantee that, but for now we didn't meet this case so I propose to keep this two values for a while.